File: backend/DataAccess/Entities/Trip.cs
using System.Diagnostics;

namespace DataAccess.Entities
{
    public class Trip
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string? Description { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public decimal PlannedBudget { get; set; }
        public string? GeneralNotes { get; set; }
        public int OwnerUserId { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        // Navigation
        public User Owner { get; set; } = null!;
        public ICollection<Destination> Destinations { get; set; } = new List<Destination>();
        public ICollection<Activity> Activities { get; set; } = new List<Activity>();
        public ICollection<Expense> Expenses { get; set; } = new List<Expense>();
        public ICollection<ChecklistItem> ChecklistItems { get; set; } = new List<ChecklistItem>();
        public ICollection<ShareToken> ShareTokens { get; set; } = new List<ShareToken>();
    }
}

File: frontend/src/models/Trip.ts
import type { Activity } from "./Activity";
import type { ChecklistItem } from "./ChecklistItem";
import type { Destination } from "./Destination";
import type { Expense } from "./Expense";


export interface Trip {
  id: number;
  name: string;
  description?: string;
  startDate: string;          
  endDate: string;
  plannedBudget: number;
  generalNotes?: string;
  destinationCount: number;
  totalSpent: number;
}

export interface TripDetail {
  id: number;
  name: string;
  description?: string;
  startDate: string;
  endDate: string;
  plannedBudget: number;
  generalNotes?: string;
  ownerUserId: number;
  destinations: Destination[];
  activities: Activity[];
  expenses: Expense[];
  checklistItems: ChecklistItem[];
  budgetSummary: BudgetSummary;
}

export interface CreateTripRequest {
  name: string;
  description?: string;
  startDate: string;
  endDate: string;
  plannedBudget: number;
  generalNotes?: string;
}

export interface UpdateTripRequest {
  name: string;
  description?: string;
  startDate: string;
  endDate: string;
  plannedBudget: number;
  generalNotes?: string;
}

export interface BudgetSummary {
  plannedBudget: number;
  totalSpent: number;
  remaining: number;
}